### PR TITLE
Additions to Customer Logging & Monitoring

### PR DESCRIPTION
**Pull request by charlesakalugwu**
This adds some further pieces to the Customer Logging and Monitoring work.

```release-note
NONE
```

/cc @asalkeld @ehashman @kwoodson @thekad 
/hold

### CODE REVIEW

File: pkg/sync/v8/data/ConfigMap/openshift-azure-logging/log-analytics-agent-entrypoint.yaml [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    REGULAR_USER=omsagent
+
+    if [ -S ${DOCKER_SOCKET} ]; then
+        echo "getting gid for docker.sock"
+        DOCKER_GID=$(stat -c '%g' ${DOCKER_SOCKET})
+        echo "adding omsagent user to docker group ${DOCKER_GID}"
+        cat >>/etc/group <<EOL
+    ${DOCKER_GROUP}:x:${DOCKER_GID}:${REGULAR_USER}
+    EOL
+    fi
```
Comment (charlesakalugwu) [inline pkg/sync/v8/data/ConfigMap/openshift-azure-logging/log-analytics-agent-entrypoint.yaml]: This if block is the only place where main.sh is changed. Compare it to https://github.com/microsoft/OMS-docker/blob/2b721554eff9ed9108368f7ac959ece589e9f7e9/ci_feature_prod/main.sh#L32-L39

File: pkg/sync/v8/data/Deployment.apps/openshift-azure-logging/log-analytics-cluster-agent.yaml [hunk 2]
```diff
@@ -82,7 +82,7 @@ spec:
           subPath: main.sh
       nodeSelector:
         beta.kubernetes.io/os: linux
-        node-role.kubernetes.io/infra: "true"
+        node-role.kubernetes.io/compute: "true"
```
Comment (asalkeld) [inline pkg/sync/v8/data/Deployment.apps/openshift-azure-logging/log-analytics-cluster-agent.yaml]: I set this infra as it's a deployment (one per cluster) and the machines on infra are well suited to this (lots of memory).
Also I don't believe it makes a difference as where this runs (as far as it's functionality goes), it is getting inventory.

File: pkg/sync/v8/data/ConfigMap/openshift-azure-logging/log-analytics-agent-entrypoint.yaml (hunk 1 above)
Comment (asalkeld) [reply pkg/sync/v8/data/ConfigMap/openshift-azure-logging/log-analytics-agent-entrypoint.yaml]: here is a way to reduce the copying : https://github.com/openshift/openshift-azure/commit/cb4e803fd48b9ed8abaea3b6252695e1e0f0a692

File: pkg/sync/v8/data/Deployment.apps/openshift-azure-logging/log-analytics-cluster-agent.yaml (hunk 2 above)
Comment (charlesakalugwu) [reply pkg/sync/v8/data/Deployment.apps/openshift-azure-logging/log-analytics-cluster-agent.yaml]: i'm ok with this

File: pkg/sync/v8/data/ConfigMap/openshift-azure-logging/log-analytics-agent-entrypoint.yaml (hunk 1 above)
Comment (charlesakalugwu) [reply pkg/sync/v8/data/ConfigMap/openshift-azure-logging/log-analytics-agent-entrypoint.yaml]: incorporated your suggestion

**Pull request merged**